features/mesh/dataplanes/Index.feature

```diff
@@ -120,11 +98,112 @@ Feature: mesh / dataplanes / index
     When I visit the "/meshes/default/data-planes" URL
 
     Then the "$service-cell" element is empty
-    Then the "$protocol-cell" element is empty
-    Then the "$zone-cell" element is empty
 
     Then the "$item:nth-child(1)" element contains
       | Value          |
       | dpp-2          |
       | No certificate |
       | offline        |
+
+  Rule: The listing can be filtered by type
+    Scenario: Filtering by "builtin"
+      Given the environment
+        """
+        KUMA_DATAPLANE_COUNT: 1
+        KUMA_DATAPLANEINBOUND_COUNT: 0
+        """
+      And the URL "/meshes/default/dataplanes/_overview" responds with
+        """
+        body:
+          items:
+            - name: fake-transmitter-gateway_builtin-0
+              dataplane:
+                networking:
+                  gateway:
+                    type: 'BUILTIN'
+                    tags: {}
```

Comment:
This can’t be filled out by mocks reliably right now. When we generate the mocks for a list, the default mocks will have random Dataplane types. We do mock out the first element here to have the type we expect, but the original mock might not have been a gateway type and therefore doesn’t always have `gateway.tags`. I updated this test to actually set a `kuma.io/service` tag here so it’s not confusing.

Reply:
I did update the mocks slightly to always produce service tags in use cases where that’s expected, but that doesn’t actually help here.